tools/pot/openvino/tools/pot/graph/model_utils.py

```diff
@@ -62,13 +62,23 @@ def get_nodes_by_type(model: NXModel, types: list):
             for node in ge.get_nodes_by_type(model_dict['model'], types)]
 
 
+def get_nodes_by_type_recursively(model: NXModel, types: list):
+    """ Returns all nodes with type from types collection
+    :param model: NXModel model
+    :param types: list of required types
+    :return list of nodes filtered by 'types' collection
+    """
+    return [node for model_dict in model.models
+            for node in ge.get_nodes_by_type_recursively(model_dict['model'], types)]
+
+
 def get_node_by_name(model: NXModel, name: str) -> Node:
```

Comment:
Why we've added `get_nodes_by_type_recursively` instead of updating `get_nodes_by_type` method as it was done for `get_node_by_name`?

Reply:
We need `get_nodes_by_type` without recursion in `set_rescaling_factors`, `fill_output_nodes` and  to get `Parameter` nodes only from the main graph

Reply:
In this case, I think, we can create a new method for the `Parameter` search at the main graph and change the `get_nodes_by_type` signature to reduce the number of changes. But, not sure, need comments from @AlexKoff88

Reply:
In this case, it is more a matter of taste. Probably, we can have only one function, e.g. `def get_nodes_by_type(graph: Graph, types: list, recurively=True):`, but it does not lead to code reuse or minimization. The only benefit I see, it will help to have the same code and the name of the function is smaller. But maybe it makes sense since this will minimize the code diff.
@l-bat, can we revise the code if it is quick?

Reply:
Shouldn't we do the same thing for `get_node_by_name(graph: Graph, name: str, recurively=True)` and `def get_all_operation_nodes(graph: Graph, recurively=True)` to make everything consistent?